src/NuGet.Status/Controllers/AdminController.cs

```diff
@@ -3,94 +3,119 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
 using System.Runtime.CompilerServices;
 using System.Threading.Tasks;
 using System.Web.Mvc;
 using Microsoft.WindowsAzure.Storage.Table;
 using NuGet.Services.Status.Table.Manual;
+using NuGet.Status.Configuration;
 using NuGet.Status.Helpers;
 using NuGet.Status.Models;
 using NuGet.Status.Utilities;
 
 namespace NuGet.Status.Controllers
 {
-    [Authorize(Roles = "Admins")]
+    [Authorize]
     public class AdminController : AppController
     {
         public const string UpdatedTempDataKey = "Updated";
 
+        private string[] adminIdentities;
+
+        public AdminController()
+        {
+            adminIdentities = MvcApplication.StatusConfiguration?.AdminIdentities?.Split(';') ??
+                throw new HttpRequestException("Authorized user configuration failure.");
+        }
+
         [HttpGet]
         public ActionResult Index()
         {
-            return View(nameof(Index), CachedServiceStatus?.ServiceStatus);
+            return AuthorizedAction(() => 
```

Comment:
I don't like that you have to remember to call `AuthorizedAction` here. What if we add a new action and forget to wrap the logic in `AuthorizedAction`? Wouldn't that allow users not in the admin identity group to perform the action?

Can we have a custom filter applied to the controller that verifies this instead of putting the logic in the action (method) body?

Reply:
Interesting idea. I can explore that.

Reply:
Remember that this is intended to be temporary though. But those are famous last words. :)

Reply:
Okay, yes, this idea is much nicer. Pushing the change.